cluster-autoscaler/cloudprovider/mcm/mcm_manager.go

```diff
@@ -508,27 +474,32 @@ func (m *McmManager) DeleteMachines(targetMachineRefs []*Ref) error {
 	if !isRollingUpdateFinished(md) {
 		return fmt.Errorf("MachineDeployment %s is under rolling update , cannot reduce replica count", commonMachineDeployment.Name)
 	}
+	markedMachines := getMachinesMarkedByCAForDeletion(md)
 	// update priorities of machines to be deleted except the ones already in termination to 1
-	scaleDownAmount, err := m.prioritizeMachinesForDeletion(targetMachineRefs)
+	machinesWithPrio1, err := m.prioritizeMachinesForDeletion(targetMachineRefs)
 	if err != nil {
 		return err
 	}
+	markedMachines.Insert(machinesWithPrio1...)
 	// Trying to update the machineDeployment till the deadline
 	err = m.retry(func(ctx context.Context) (bool, error) {
```

Comment:
Please consider whether one of the apimachinery `wait.Poll*` utility functions can be used or whether `retry` can delete to one of the apimachinery `wait.Poll*` functions. `wait.PollUntilContextTimeout` seems suitable.

Reply:
We are using our retry since `wait.PollUntilContextTimeout` does not give logs. We were able to debug the issue thanks to these logs so I think we should still keep it